tempesta_fw/ss_skb.c

```diff
@@ -744,21 +755,33 @@ ss_skb_cutoff_data(const TfwStr *hdr, int skip, int tail)
 			continue;
 		}
 		it = it_zero;
-		r = __skb_fragment(c->skb, NULL,
-				   (char *)c->ptr + skip, skip - c->len, &it);
-		if (r)
+		f_len = skip - c->len;
+		f_ptr = (char *)c->ptr + skip;
+		r = __skb_fragment(c->skb, NULL, f_ptr, f_len, &it);
+		if (r < 0)
 			return r;
+		BUG_ON(r != -f_len);
 		skip = 0;
 	}
 
 	/* Cut off the tail. */
 	if (tail) {
+		struct sk_buff *f_skb;
 		BUG_ON(it.ptr == NULL);
 		BUG_ON(it.skb == NULL);
-		r = __skb_fragment(it.skb, NULL, it.ptr, -tail, &it_zero);
-		if (r)
-			SS_WARN("Cannot delete hdr tail\n");
-		return r;
+		while (tail) {
+			f_ptr = it.ptr;
+			f_skb = it.skb;
+			it = it_zero;
```

Comment:
Yes, but `it` values are checked on return from `__split_linear_data()` or `__split_pgfrag()`, and that depends on tha fact that `it` values **may** have changed if they had been set. To see if they have changed, we need to have the initial values for `it` member zeroed.

I will use `memset()` instead.